vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionManager.java

```diff
@@ -120,10 +126,43 @@ public void updateExhibitionLanguages(Exhibition exhibition, List<String> codes,
                 exhibitionLanguage.setDefault(exhibitionLanguage.getCode().equalsIgnoreCase(defaultLanguage));
             });  
 
-        // Removes exhibition langauge if unselected.
-        exhibition.getLanguages().removeAll(exhibition.getLanguages().stream()
-                .filter(language -> !codes.contains(language.getCode())).collect(Collectors.toList()));
+        // Finds exhibition language if unselected (to be deleted).
+        List<IExhibitionLanguage> exhibitionLanguageToBeRemoved = exhibition.getLanguages().stream()
+                .filter(language -> !codes.contains(language.getCode())).collect(Collectors.toList());
+
+        for (IExhibitionLanguage language  : exhibitionLanguageToBeRemoved ) {
+            if(checkIfLocalizedTextExists(language))  {
+                throw new ExhibitionLanguageDeletionException() ;
+            }
+            language.getLocalizedTexts().clear();
+            //delete the empty localized texts
+            localizedTextRepo.deleteAll(language.getLocalizedTexts());
+        }
+
+        
+        exhibition.getLanguages().removeAll(exhibitionLanguageToBeRemoved);
+
+    }
+    
+    /**
+     * Return true if given exhibition language has non empty localized texts linked to it
+     * 
+     */
+    @Override
+    public boolean checkIfLocalizedTextExists(IExhibitionLanguage language)  {        
+
+        return !CollectionUtils.isEmpty(language.getLocalizedTexts()) || !CollectionUtils.isEmpty(language.getLocalizedTexts()) && !checkIfTextIsEmpty(language.getLocalizedTexts());
+
+    }
+    
+    /**
+     * 
+     * @param localizedTexts
+     * @return
+     */
+    private boolean checkIfTextIsEmpty(List<LocalizedText> localizedTexts) {
```

Comment:
this method is used on a list of texts, right? So it should be `checkIfTextsAreEmpty`